### PR TITLE
Develop feature/olp 1252

### PR DESCRIPTION
**Pull request by ws4charlie**
1. Implemented 'Reinvest Rewards' transaction.
2. Integration test for ReinvestRewards transaction.
3. Defined pool names as constants and unified the way we get pool address by its name

### CODE REVIEW

File: action/network_delegation/reinvest_rewards.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	//Add Delegation
+	//Get Delegation Pool
+	poolList, err := ctx.GovernanceStore.GetPoolList()
+	if err != nil {
+		return helpers.LogAndReturnFalse(ctx.Logger, gov.ErrPoolList, invest.Tags(), err)
+	}
+	if _, ok := poolList["DelegationPool"]; !ok {
+		return helpers.LogAndReturnFalse(ctx.Logger, action.ErrPoolDoesNotExist, invest.Tags(), err)
+	}
+	delagationPool := poolList["DelegationPool"]
```
Comment (kingpinXD) [inline action/network_delegation/reinvest_rewards.go:130]: We should probably add the strings as constants , probably in data/governanace/store.go

File: service/tx/network_delegation.go [hunk 2]
```diff
@@ -109,6 +109,38 @@ func (s *Service) WithdrawDelegRewards(args client.WithdrawDelegRewardsRequest,
...
+	}
+
+	uuidNew, _ := uuid.NewUUID()
+	feeAmount := s.feeOpt.MinFee()
+	tx := &action.RawTx{
+		Type: action.REWARDS_REINVEST_NETWORK_DELEGATE,
+		Data: data,
+		Fee: action.Fee{
+			Price: action.Amount{Currency: "OLT", Value: *feeAmount.Amount},
+			Gas:   80000,
```
Comment (kingpinXD) [inline service/tx/network_delegation.go:130]: Just curious ,does the trasaction fail with not enough gas ,for values less than 80000  ?

Comment (ws4charlie) [reply service/tx/network_delegation.go:130]: > Just curious ,does the trasaction fail with not enough gas ,for values less than 80000 ?

I don't know the exact number of min gas that cause a failure, 80000 is an estimated gas big enough to guarantee a success.

Comment (kerhbal) [reply service/tx/network_delegation.go:130]: I failed at 40000 for some txs before

File: action/network_delegation/reinvest_rewards.go (hunk 1 above)
Comment (ws4charlie) [reply action/network_delegation/reinvest_rewards.go]: > We should probably add the strings as constants , possibly in data/governanace/store.go

Good idea to unify these.

**Review by kingpinXD — DISMISSED** [review]

**Review by kingpinXD — APPROVED** [review]

**Review by kingpinXD — APPROVED** [review]

**Review by kingpinXD — APPROVED** [review]

**Pull request merged**